--- origsrc/GConf-2.22.0/backends/xml-backend.c	2007-06-25 06:11:19.000000000 -0500
+++ src/GConf-2.22.0/backends/xml-backend.c	2008-05-28 07:37:24.796875000 -0500
@@ -797,7 +797,7 @@
 
 /* Initializer */
 
-#ifndef G_OS_WIN32
+#ifndef G_PLATFORM_WIN32
 /* If we use G_MODULE_EXPORT, *only* thusly marked functions will be
  * exported, and xml-test uses other ones, too.
  */
@@ -814,7 +814,7 @@
   return NULL;
 }
 
-#ifndef G_OS_WIN32
+#ifndef G_PLATFORM_WIN32
 G_MODULE_EXPORT
 #endif
 GConfBackendVTable* 
